code/parachain/frame/composable-support/src/collections/vec/bounded/bi_bounded_vec.rs

```diff
@@ -55,7 +75,65 @@ pub enum BiBoundedVecOutOfBounds {
 	},
 }
 
+impl<T, const U: usize> BiBoundedVec<T, 1, U> {
+	pub fn new(first: T) -> Self {
+		BiBoundedVec::from_vec(vec![first]).expect("compile time verified")
+	}
+}
+
+impl<T, const U: usize> BiBoundedVec<T, 0, U> {
+	pub fn first(&self) -> Option<&T> {
+		self.inner.first()
+	}
+}
+
+impl<T, const U: usize> BiBoundedVec<T, 2, U> {
+	pub fn new(first: T, second: T) -> Self {
+		BiBoundedVec::from_vec(vec![first, second]).expect("compile time verified")
+	}
+}
+
+impl<T, const L: usize, const U: usize> BiBoundedVec<T, L, U>
+where
+	ConstAssert<{ L > 0 }>: ConstTrue,
+{
+	pub fn first_mut(&mut self) -> &mut T {
+		self.inner.get_mut(0).expect("verified by compiler")
+	}
+
+	/// Returns the first element of non-empty Vec
+	///
+	/// # Example
+	/// ```rust
+	/// use composable_support::collections::vec::bounded::BiBoundedVec;
+	/// use sp_std::convert::TryInto;
+	///
+	/// let data: BiBoundedVec<_, 2, 8> = vec![1u8, 2].try_into().unwrap();
+	/// assert_eq!(data.first(), &1);
+	/// ```
+	pub fn first(&self) -> &T {
```

Comment:
why to remove option? wouldnt this crash? dont we have that uses the previous version of first function?

Reply:
we care abound bounded case. for bounde case it is it, fomr 0 bound it is option. see 0 and > impla of first